### PR TITLE
Added a new `PrepareBackup` RPC for snapshot backup

### PR DESCRIPTION
**Pull request by YuJuncen**
### Q & A

Some questions you may like to ask:

#### Q: Is it possible to split this into two RPCs?

Surely we can split this into two RPCs: one for ready and one for wait applies.

This will allow us to reuse some gRPC's infras to control connections, and easier to be understood the behavior.

But I don't want to do this because:
- It seems waiting apply is meanless if we haven't disabled new commands proposing, which seems even harder to understand "why this RPC even exists?".
- Controlling many asynchronous requests seems not really easier than a sole connection with many tagged request / responses.

### CODE REVIEW

File: proto/brpb.proto [hunk 1]
```diff
@@ -15,6 +15,30 @@ option (rustproto.lite_runtime_all) = true;
 
 option java_package = "org.tikv.kvproto";
 
+enum PrepareSnapshotBackupRequestType {
+    UpdateLease = 0;
```
Comment (Connor1996) [inline proto/brpb.proto:19]: what's this used for?

File: proto/brpb.proto [hunk 2]
```diff
@@ -15,6 +15,30 @@ option (rustproto.lite_runtime_all) = true;
...
+    WaitApply = 1;
+}
+
+message PrepareSnapshotBackupRequest {
```
Comment (overvenus) [inline proto/brpb.proto:23]: Please add some comments about the message and fields.

File: proto/brpb.proto (hunk 1 above)
Comment (YuJuncen) [reply proto/brpb.proto:19]: Updating the TTL, I will add some comments then.

File: proto/brpb.proto [hunk 3]
```diff
@@ -15,6 +15,47 @@ option (rustproto.lite_runtime_all) = true;
 
 option java_package = "org.tikv.kvproto";
 
+enum PrepareSnapshotBackupRequestType {
+    // Update the lease of suspending new ingest / admin commands to be proposed.
```
Comment (Connor1996) [inline proto/brpb.proto:19]: Please describe what's the lease used for

**Review by Connor1996 — COMMENTED** [review]
rest LGTM

**Pull request merged**